### DISCUSSION TITLE
[Question][iOS] How should I add resources to my app?

### DISCUSSION THREAD

**Discussion by Guidemarcus** [unoplatform/uno#12209] [Q&A]
Hi,
I am adding some resources to my app and, on iOS, it throws an error "[Object must implement IConvertible". 

```c#
	Current.Resources.Add("ScreenWidth", UIKit.UIScreen.MainScreen.Bounds.Width);
```
After some investigation, I found out that adding a cast for double fixes the issue. 
```c#
	Current.Resources.Add("ScreenWidth", (double)UIKit.UIScreen.MainScreen.Bounds.Width);
```

Is this a regression (went from uno.ui to uno.winui)? Is this simply a new behavior?

**Comment by jeromelaban** [comment]
Can you provide the context of this code, particularly what is `Current` ? Can you also provide a stack trace of the exception?

**Comment by Guidemarcus** [comment]
to add to this topic. The exception is thrown when I bind the new resource to a Width property in the view
